src/ploigos_step_runner/results/workflow_result.py

```diff
@@ -239,8 +234,49 @@ def load_from_pickle_file(pickle_filename):
         except Exception as error:
             raise StepRunnerException(f'error loading {pickle_filename}: {error}') from error
 
+    def merge_with_pickle_file(self, pickle_filename):
+        """Merge our workflow list with that stored on disk.
+        When we find overlaps, our in-memory values win.
+
```

Comment:
nit: maybe add the note about responsiblity of locking, like you did in write_to_pickle